packages/picasso.js/src/core/component/component-factory.js

```diff
@@ -551,6 +557,9 @@ function componentFactory(definition, context = {}) {
 
     setUpEmitter(instanceContext, emitter, config);
     setUpEmitter(definitionContext, emitter, definition);
+
+    setVisibility(instanceContext, true);
```

Comment:
@miralemd Did you mean like this?

Reply:
almost - skip the `setVisibility` part and update `isVisible` from within `mount/unmount`, and then return that value from `ctx.isVisible()`: